dds/DCPS/security/SSL/SubjectName.cpp

```diff
@@ -145,28 +139,204 @@ namespace SSL {
 
   int SubjectName::parse_ldap_v3(const char* in)
   {
-    return simple_avp_seq_parse(in, ",", "=", " ", " ", false);
+    Parser parser(in);
+    return parser.parse(attr_vec_) ? 0 : 1;
   }
 
   bool SubjectName::operator==(const SubjectName& rhs) const
   {
-    bool result = (map_.size() == rhs.map_.size());
-    for (AttrMap::const_iterator i1 = map_.begin(), i2 = rhs.map_.begin();
-         result == true && i1 != map_.end() && i2 != rhs.map_.end();
-         ++i1, ++i2) {
-      if (i1->first.compare(i2->first) != 0 ||
-          i1->second.compare(i2->second) != 0) {
-        result = false;
+    if (attr_vec_.size() != rhs.attr_vec_.size()) {
```

Comment:
Can this should be simplified to `attr_vec_ == rhs.attr_vec_`?

Reply:
Yes, I think that works.